test/e2e/apps/deployment.go

```diff
@@ -785,6 +788,116 @@ func testDeploymentsControllerRef(f *framework.Framework) {
+	}
+
+	framework.Logf("Verifying that the replicasets of deployment %q have desired number of replicas", deploymentName)
+	firstRS, err = c.ExtensionsV1beta1().ReplicaSets(ns).Get(firstRS.Name, metav1.GetOptions{})
+	Expect(err).NotTo(HaveOccurred())
+	secondRS, err := deploymentutil.GetNewReplicaSet(deployment, c.ExtensionsV1beta1())
+	Expect(err).NotTo(HaveOccurred())
+
+	err = framework.WaitForReplicaSetDesiredReplicas(c.ExtensionsV1beta1(), firstRS)
+	Expect(err).NotTo(HaveOccurred())
+	err = framework.WaitForReplicaSetDesiredReplicas(c.ExtensionsV1beta1(), secondRS)
+	Expect(err).NotTo(HaveOccurred())
+
+	framework.Logf("Verifying that first rollout's replicaset has minimum required number of available replicas")
+	if firstRS.Status.AvailableReplicas < minAvailableReplicas {
+		Expect(fmt.Errorf("observed %d available replicas, less than min required %d", firstRS.Status.AvailableReplicas, minAvailableReplicas)).NotTo(HaveOccurred())
+	}
+
+	framework.Logf("Waiting for first rollout's replicaset to have 8 available replicas")
+	err = framework.WaitForReplicaSetAvailableReplicas(c.ExtensionsV1beta1(), firstRS, 8)
+	Expect(err).NotTo(HaveOccurred())
+
+	framework.Logf("Verifying that second rollout's replicaset does not have any available replicas")
+	Expect(secondRS.Status.AvailableReplicas).Should(Equal(int32(0)))
+
+	newReplicas := int32(15)
+	framework.Logf("Scaling up the deployment %q from %d to %d", deploymentName, replicas, newReplicas)
+	deployment, err = framework.UpdateDeploymentWithRetries(c, ns, deployment.Name, func(update *extensions.Deployment) {
+		update.Spec.Replicas = &newReplicas
+	})
+	Expect(err).NotTo(HaveOccurred())
+
+	framework.Logf("Waiting for observed generation %d", deployment.Generation)
+	Expect(framework.WaitForObservedDeployment(c, ns, deploymentName, deployment.Generation)).NotTo(HaveOccurred())
+
+	framework.Logf("Verifying that the replicasets of deployment %q have desired number of replicas", deploymentName)
+	firstRS, err = c.ExtensionsV1beta1().ReplicaSets(ns).Get(firstRS.Name, metav1.GetOptions{})
+	Expect(err).NotTo(HaveOccurred())
+	secondRS, err = c.ExtensionsV1beta1().ReplicaSets(ns).Get(secondRS.Name, metav1.GetOptions{})
+	Expect(err).NotTo(HaveOccurred())
+
+	err = framework.WaitForReplicaSetDesiredReplicas(c.ExtensionsV1beta1(), firstRS)
+	Expect(err).NotTo(HaveOccurred())
+	err = framework.WaitForReplicaSetDesiredReplicas(c.ExtensionsV1beta1(), secondRS)
+	Expect(err).NotTo(HaveOccurred())
+
+	framework.Logf("Verifying that first rollout's replicaset has 11 available replicas")
+	err = framework.WaitForReplicaSetAvailableReplicas(c.ExtensionsV1beta1(), firstRS, 11)
+	Expect(err).NotTo(HaveOccurred())
+
+	framework.Logf("Verifying that second rollout's replicaset does not have any available replicas")
+	Expect(secondRS.Status.AvailableReplicas).Should(Equal(int32(0)))
```

Comment:
Since we care about proportional scaling, it makes sense to check `spec.replicas` of the 2nd RS.

Reply:
Agreed. Should we check both `Spec.Replicas` and `.Status.AvailableReplicas` of the 2nd RS? Or is it fine to remove the `.Status.AvailableReplicas` check that I have written?